backend/src/data/repositories/user-details/user-details.repository.ts

```diff
@@ -61,6 +62,21 @@ class UserDetails {
       .execute();
   }
 
+  public updateMoneyBalance(
+    userId: number,
+    newMoneyBalance: number,
+  ): Promise<UserDetailsWithMoneyBalanceDto | null> {
```

Comment:
If you're type casting the result of query to `UserDetailsWithMoneyBalanceDto` and sure that it can't be null, then in method's contract returning type also won't be `null`
```suggestion
  ): Promise<UserDetailsWithMoneyBalanceDto> {
```